### PR TITLE
Map rotation touchup, automatic map votes

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
I heard rumblings about people abusing restart votes and shared vote cooldown to block map votes (again), so I guess we can't leave this to the players. If the escape shuttle leaves and there hasn't been a map vote yet, a map vote will automatically be started. If the round ends without a map vote (ie from non continuous game modes), map rotation will weighted pick a new map based on player preferences. Map rotation now takes the same map blocking rules as map votes into account.
Note that any form of setting next map will stop the automatic map vote and rotation, so admins can still have the final word if they wish.

While at it, increased the stored rounds to 3. This means that any map that has occured twice in the last three + current rounds is not present in map rotation or map votes.

Improved the code side of this slightly, makes the file saving proc properly compatible with changing stored length as it wouldn't have been able to reduce the rounds stored. 

## Why It's Good For The Game
More consistent map diversity, blocks vote abuse.

## Changelog
:cl: Skoglol
tweak: Map votes and rotation tweaked further. Map vote will automatically run as the shuttle leaves, and if no map vote has been run by round end map rotation will trigger.
tweak: The last 3 maps are now stored, up from 2. This means any map run twice in the last four (including current) rounds are disqualified from map vote and map rotation.
/:cl:

### CODE REVIEW

**Comment by Dorsisdwarf** [comment]
Does this take into account the current limits on map rotation (AKA it only triggers after reasonable length rounds so the station doesnt change when Sudden Death the wizard teleports into the armory and gets filled with lead 15 seconds in)?

**Comment by kriskog** [comment]
No, it doesn't. Instant round enders like wizard dying will trigger map rotation no matter how long the  round was, and I don't see why it shouldn't. I'll sleep on it.

**Comment by cacogen** [comment]
love u

**Comment by kriskog** [comment]
Uh forgot to say, now doesnt force rotate on very short (<10 minute) rounds. If a vote has been passed that still applies though.

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -116,6 +119,18 @@ SUBSYSTEM_DEF(persistence)
...
+	//Convert the mapping data to a shared blocking list, saves us doing this in several places later.
+	for(var/map in config.maplist)
+		var/datum/map_config/VM = config.maplist[map]
+		var/run = 0
+		if(VM.map_name == SSmapping.config.map_name)
+			run++
+		for(var/name in SSpersistence.saved_maps)
+			if(VM.map_name == name)
+				run++
+		if(run >= 2) //If run twice in the last five (including current) rounds as of time of writing, disable map for voting and rotation.
```
Comment (ninjanomnom) [inline code/controllers/subsystem/persistence.dm]: This comment seems to be a lie already

Comment (kriskog) [reply code/controllers/subsystem/persistence.dm]: Ahem, yes. Woops

**Review by ninjanomnom — APPROVED** [review]

**Comment by wejengin2** [comment]
i like this

**Pull request merged**